cpp/src/io/utilities/config_utils.hpp

```diff
@@ -23,7 +24,17 @@ namespace cudf::io::detail {
  * @brief Returns the value of the environment variable, or a default value if the variable is not
  * present.
  */
-std::string getenv_or(std::string const& env_var_name, std::string_view default_val);
```

Comment:
I feel something dangerous here. When you call `getenv_or("abc")`, you construct a temp string then in turn construct a `string_view` from it. So you are viewing a temp object which may not be guaranteed to exist in this function.

Reply:
I thought that was okay, [per cppref](https://en.cppreference.com/w/cpp/language/lifetime)
>All temporary objects are destroyed as the last step in evaluating the full-expression that (lexically) contains the point where they were created, and if multiple temporary objects were created, they are destroyed in the order opposite to the order of creation. This is true even if that evaluation ends in throwing an exception. 

Please correct me if wrong.

Reply:
Sorry I'm not sure what you mean?
To clarify what I said, the `string_view` parameter does not own the temp string. As such, the temp string may be destroyed immediately before entering the function.

Reply:
Okay, thank you. So it is safe to use in this case. I also found a long explanation for it here: https://www.foonathan.net/2017/03/string_view-temporary/

Reply:
I search over the internet and found a huge collection of the same topic: Is it safe to use `string_view` on temp string?

Reply:
AFAICT using `string_view` has the same pitfalls as const&. It should be fine as a parameter. I can change it back, don't have a strong preference.

Reply:
It's fine for me now.